lib/std/io/tty.zig

```diff
@@ -7,37 +7,41 @@ const native_os = builtin.os.tag;
 
 /// Detect suitable TTY configuration options for the given file (commonly stdout/stderr).
 /// This includes feature checks for ANSI escape codes and the Windows console API, as well as
-/// respecting the `NO_COLOR` environment variable.
+/// respecting the `NO_COLOR` and `YES_COLOR` environment variables to override the default.
 pub fn detectConfig(file: File) Config {
-    return detectConfigForce(file, false);
+    const force_color: ?bool = if (builtin.os.tag == .wasi)
+        null // wasi does not support environment variables
+    else if (process.hasEnvVarConstant("NO_COLOR"))
+        false
+    else if (process.hasEnvVarConstant("YES_COLOR"))
+        true
+    else
+        null;
+    return detectConfigForce(file, force_color);
 }
 
-/// Detect suitable TTY configuration options, but force the use of color if the platform
-/// supports it. On platforms such as wasi that do not support ANSI codes, if they are
-/// desired for some reason just use `Config.escape_codes` directly.
-pub fn detectConfigForce(file: File, force_color: bool) Config {
-    if (builtin.os.tag == .wasi) {
-        // Per https://github.com/WebAssembly/WASI/issues/162 ANSI codes
-        // aren't currently supported.
-        return .no_color;
-    } else if (process.hasEnvVarConstant("ZIG_DEBUG_COLOR")) {
-        return .escape_codes;
-    } else if (process.hasEnvVarConstant("NO_COLOR") and !force_color) {
-        return .no_color;
-    } else if (file.supportsAnsiEscapeCodes()) {
-        return .escape_codes;
-    } else if (native_os == .windows and file.isTty()) {
+/// Detect suitable TTY configuration options, along with an override parameter.
+/// Even if `force_color` is `true`, this will prefer Windows console API calls
+/// if `file` is a Windows terminal.
+pub fn detectConfigForce(file: File, force_color: ?bool) Config {
+    if (force_color == false) return .no_color;
+
+    if (native_os == .windows and file.isTty()) {
         var info: windows.CONSOLE_SCREEN_BUFFER_INFO = undefined;
         if (windows.kernel32.GetConsoleScreenBufferInfo(file.handle, &info) != windows.TRUE) {
-            // TODO: Should this return an error instead?
-            return .no_color;
+            return if (force_color == true) .escape_codes else .no_color;
```

Comment:
I wasn't actually sure how windows works with colour so I left the previous behaviour. If windows would support escape codes here this seems fine, if not similar to WASI I'd suggest returning `.no_color` unconditionally.